build.rs

```diff
@@ -119,41 +121,120 @@ fn main() {
+                    "arch/ARM/ARMModule.c",
+                    "cs.c",
+                    "MCInst.c",
+                    "MCInstrDesc.c",
+                    "MCRegisterInfo.c",
+                    "arch/Mips/MipsDisassembler.c",
+                    "arch/Mips/MipsInstPrinter.c",
+                    "arch/Mips/MipsMapping.c",
+                    "arch/Mips/MipsModule.c",
+                    "arch/PowerPC/PPCDisassembler.c",
+                    "arch/PowerPC/PPCInstPrinter.c",
+                    "arch/PowerPC/PPCMapping.c",
+                    "arch/PowerPC/PPCModule.c",
+                    "arch/Sparc/SparcDisassembler.c",
+                    "arch/Sparc/SparcInstPrinter.c",
+                    "arch/Sparc/SparcMapping.c",
+                    "arch/Sparc/SparcModule.c",
+                    "SStream.c",
+                    "arch/SystemZ/SystemZDisassembler.c",
+                    "arch/SystemZ/SystemZInstPrinter.c",
+                    "arch/SystemZ/SystemZMapping.c",
+                    "arch/SystemZ/SystemZMCTargetDesc.c",
+                    "arch/SystemZ/SystemZModule.c",
+                    "utils.c",
+                    "arch/X86/X86ATTInstPrinter.c",
+                    "arch/X86/X86Disassembler.c",
+                    "arch/X86/X86DisassemblerDecoder.c",
+                    "arch/X86/X86IntelInstPrinter.c",
+                    "arch/X86/X86Mapping.c",
+                    "arch/X86/X86Module.c",
+                    "arch/XCore/XCoreDisassembler.c",
+                    "arch/XCore/XCoreInstPrinter.c",
+                    "arch/XCore/XCoreMapping.c",
+                    "arch/XCore/XCoreModule.c",
+                ];
+
+                cc::Build::new()
+                    .files(files.into_iter().map(|f| format!("{}/{}", CAPSTONE_DIR, f)))
+                    .include("capstone/include")
+                    .define("CAPSTONE_X86_ATT_DISABLE_NO", None)
+                    .define("CAPSTONE_DIET_NO", None)
+                    .define("CAPSTONE_X86_REDUCE_NO", None)
+                    .define("CAPSTONE_HAS_ARM", None)
+                    .define("CAPSTONE_HAS_ARM64", None)
+                    .define("CAPSTONE_HAS_MIPS", None)
+                    .define("CAPSTONE_HAS_POWERPC", None)
+                    .define("CAPSTONE_HAS_SPARC", None)
+                    .define("CAPSTONE_HAS_SYSZ", None)
+                    .define("CAPSTONE_HAS_X86", None)
+                    .define("CAPSTONE_HAS_XCORE", None)
+                    .define("CAPSTONE_USE_SYS_DYN_MEM", None)
+                    .flag_if_supported("-Wno-unused-function")
+                    .flag_if_supported("-Wno-unused-parameter")
+                    .flag_if_supported("-Wno-unknown-pragmas")
+                    .flag_if_supported("-Wno-sign-compare")
+                    .static_crt(use_static_crt)
+                    .compile("capstone");
+            }
+
+            #[cfg(not(windows))]
+            {
```

Comment:
Could we also refactor this block into a separate function? Something like `build_capstone_gmake()`?